lib/schema.ts

```diff
@@ -114,18 +121,18 @@ function buildFields(schema, rLevelParentMax, dLevelParentMax, path) {
               opts.fields,
               rLevelMax,
               dLevelMax,
-              path.concat([name]))
+              path.concat(name))
       };
-
-      if (opts.type == 'LIST' || opts.type == 'MAP') fieldList[name].originalType = opts.type;
-
+      
       continue;
     }
 
+    let typeDef ;
     /* field type */
-    const typeDef = parquet_types.PARQUET_LOGICAL_TYPES[opts.type];
-    if (!typeDef) {
+    if (!opts.type) {
       throw 'invalid parquet type: ' + opts.type;
+    } else {
+      typeDef = parquet_types.PARQUET_LOGICAL_TYPES[opts.type];
```

Comment:
I think you need both checks here. You are right that `opts.type` might not exist, but even if it does, we cannot know that it does exist inside of `PARQUET_LOGICAL_TYPES`

```
    const typeDef = opts.type ? parquet_types.PARQUET_LOGICAL_TYPES[opts.type] : undefined;
    if (!typeDef) {
      throw 'invalid parquet type: ' + (opts.type || "missing type");
    }
```